From d0562705bcd4cb9849156f095b2af0ec1bb53b56 Mon Sep 17 00:00:00 2001
From: Lech Perczak <lech.perczak@gmail.com>
Date: Fri, 17 Nov 2023 21:33:04 +0100
Subject: [PATCH] ARM: dts: nxp: imx7d-pico: add cpu-supply nodes

The PICO-IMX7D SoM has the usual power supply configuration using
output sw1a of PF3000 PMIC, which was defined in downstream derivative
of linux-imx (see link) in the sources for "Android Things" devkit.
It is required to support CPU frequency scaling.

Map the respective "cpu-supply" nodes of each core to sw1a of the PMIC.

Enabling them causes cpufreq-dt, and imx-thermal drivers to probe
successfully, and CPU frequency scaling to function.

Link: https://android.googlesource.com/platform/hardware/bsp/kernel/nxp/imx-v4.1/+/o-iot-preview-5/arch/arm/boot/dts/imx7d-pico.dtsi#849

Cc: Fabio Estevam <festevam@gmail.com>
Cc: Shawn Guo <shawnguo@kernel.org>
Cc: Sascha Hauer <s.hauer@pengutronix.de>

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>
---
 arch/arm/boot/dts/imx7d-pico.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/arch/arm/boot/dts/imx7d-pico.dtsi
+++ b/arch/arm/boot/dts/imx7d-pico.dtsi
@@ -108,6 +108,14 @@
 	assigned-clock-rates = <0>, <32768>;
 };
 
+&cpu0 {
+	cpu-supply = <&sw1a_reg>;
+};
+
+&cpu1 {
+	cpu-supply = <&sw1a_reg>;
+};
+
 &ecspi3 {
 	cs-gpios = <&gpio4 11 GPIO_ACTIVE_LOW>;
 	pinctrl-names = "default";
